bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/RefreshingOAuth2CredentialsInterceptor.java

```diff
@@ -86,12 +85,12 @@ public HeaderCacheElement(IOException exception) {
 
     public CacheState getCacheState() {
       long now = System.currentTimeMillis();
-      if (now > expiresTimeMs) {
-        return CacheState.Expired;
-      } else if (now > staleTimeMs) {
+      if (now < staleTimeMs){
+        return CacheState.Good;
```

Comment:
I see, it looks like staleTimeMs < expiresTimeMs, but there is nothing in this code guarantee that is the case.
Can we at least do a check in the constructor?